### This Notbook is my personal learning journey with ChatGPT

### 📌 Step 3: Variables (Very Important!)

Variables store information.

#### Exercise 2

Create variables for:

* Your name
* Your age
* Your favorite number

Then print them.

In [7]:
# Exercise 2
name = "M Abid"
age = 25
fav_num = 27

print(name)
print(age)
print(fav_num)

M Abid
25
27


Excellent work, Abid! 👏🔥

You did everything correctly.

#### Exercise 2 Review

```python
name = "M Abid"
age = 25
fav_number = 27

print(name)
print(age)
print(fav_number)
```

✔ Variables created correctly
✔ Correct data types (string + integers)
✔ Correct printing

Very good start

### 🚀 Step 4: Numbers & Math in Python

Python can do math like a calculator.

## Basic Operators

| Symbol | Meaning        |
| ------ | -------------- |
| `+`    | Addition       |
| `-`    | Subtraction    |
| `*`    | Multiplication |
| `/`    | Division       |
| `**`   | Power          |
| `%`    | Remainder      |


#### 🧠 Bonus Challenge (Important for problem solving)

Try this:

```python
num = 7
```

Print:

* Double of the number
* Triple of the number
* Square of the number

In [12]:
# Exercise 3
x = 15
y = 4

print(x + y)
print(x - y)
print(x * y)
print(x / y)
print(x % y)

19
11
60
3.75
3


In [9]:
# Bonus Challenge
num = 7

print(num * 2)
print(num * 3)
print(num ** 2)

14
21
49


#### Exercise 3 Review

```python
x = 15
y = 4
print(x + y)
print(x - y)
print(x * y)
print(x / y)
print(x % y)
```

✔ Addition → 19
✔ Subtraction → 11
✔ Multiplication → 60
✔ Division → 3.75
✔ Remainder → 3

Perfect 👌

#### Bonus Review

```python
num = 7
print(num * 2)
print(num * 3)
print(num ** 2)
```

✔ Double → 14
✔ Triple → 21
✔ Square → 49

Excellent thinking

### 🚀 Step 5: User Input (Now it gets interesting!)

Now your program can **talk to the user**.

⚠ Important:
`input()` always gives you TEXT (string).

If you want a number, you must convert it:


#### Exercise 4

1. Ask the user for their name.
2. Ask the user for their age.
3. Print:

   ```
   Hello Abid, next year you will be 26
   ```

(Hint: You need `int()` for age.)

#### 🎯 Mini Challenge (Think a little!)

Make a small program that:

* Asks for two numbers
* Prints their sum

Example output:

```
Enter first number: 5
Enter second number: 3
The sum is 8
```

In [13]:
# Exercise 4
name = input("What is your name? ")
age = int(input("What is your age? "))

print("Hello", name, "next year you will be", age + 1)

Hello M Abid next year you will be 26


In [15]:
# Mini Challenge
first_number = int(input("Enter first number: "))
second_number = int(input("Enter second number: "))
total = first_number + second_number

print("The sum is:", total)

The sum is: 8


Let’s review carefully:

#### Exercise 4 Review

```python
name = input("What is your name? ")
age = int(input("Enter you age: "))
print("Hello", name, "next year you will be", age + 1)
```

🟢 Very good:

* Correct use of `input()`
* Correct use of `int()`
* Correct printing

#### Mini Challenge Review

```python
first_number = int(input("Enter first number: "))
second_number = int(input("Enter second number: "))
total = first_number + second_number
print("The sum is ", total)
```

✔ Perfect logic
✔ Correct conversion
✔ Correct addition

### 🚀 Step 6: If Statements (Now we enter real programming)

This is where logic begins.

Example:

```python
age = 18

if age >= 18:
    print("You are an adult")
```

Structure:

```python
if condition:
    code
```

⚠ Indentation (spaces) is VERY important in Python.

#### Exercise 5

Create a program that:

1. Asks the user for their age.
2. If age is 18 or older → print `"You can vote"`
3. Otherwise → print `"You cannot vote"`

(Hint: You need `if` and `else`)

#### Bonus Challenge

Ask the user for a number.
If the number is even → print `"Even number"`
If the number is odd → print `"Odd number"`

(Hint: Use `%`)

In [3]:
# Exercise 5
age = int(input("What is your age? "))

if age >= 18:
    print("You can vote")
else:
    print("You cannot vote")

You can vote


In [7]:
# Bouns Challenge
user_num = int(input("Enter a number: "))

if user_num %2 == 0:
    print("Odd number")
else:
    print("Even number")

Odd number


Nice work again, Abid 👏🔥
Your logic is getting stronger.

#### Exercise 5 (Voting)

```python
age = int(input("Enter your age: "))
if age >= 18:
    print("You can vote")
else:
    print("You cannot vote")
```

✔ Correct condition
✔ Correct `if/else` structure
✔ Proper indentation

Perfect ✅

#### 🔎 Bonus Challenge Review (Even / Odd)

Your code:

```python
user_num = int(input("Enter your number: "))
if user_num %2 == 0:
    print("Odd number")
else:
    print("Even number")
```

You made a **small logical mistake**:

`user_num % 2 == 0` means the number is EVEN.

But you printed `"Odd number"`.

#### Correct Version:

```python
user_num = int(input("Enter your number: "))
if user_num % 2 == 0:
    print("Even number")
else:
    print("Odd number")
```

#### Why?

* `%` gives remainder
* Even numbers divided by 2 have remainder 0
* Odd numbers have remainder 1

Example:

* `8 % 2 = 0` → Even
* `7 % 2 = 1` → Odd


### 🚀 Step 7: Multiple Conditions (elif)

Now we level up again.

#### Exercise 6

Create a program that:

1. Asks the user for a number.
2. If number is:

   * Greater than 0 → print `"Positive"`
   * Less than 0 → print `"Negative"`
   * Equal to 0 → print `"Zero"`

Use `if`, `elif`, and `else`.

#### 🎯 Mini Challenge (Logic Builder)

Make a small calculator that:

* Asks for two numbers
* Asks for an operator (`+`, `-`, `*`, `/`)
* Performs the correct calculation using `if/elif`

Example:

```
Enter first number: 5
Enter operator: *
Enter second number: 3
Result: 15
```

In [ ]:
# Exercise 6
number = int(input("Enter your number "))
if number > 0:
    print("Possitive number")
elif number < 0:
    print("Negative number")
else:
    print("Zero number")

Negative number


In [23]:
# Mini Challenge
# Basic Math Operators (+ - * /)

first_number = int(input("Enter first number: "))
second_number = int(input("Enter second number"))
operator = input("Chose one operator: ")

if operator == "+":
    print("Result:", first_number + second_number)
elif operator == "-":
    print("Result:", first_number - second_number)
elif operator == "*":
    print("Result:", first_number * second_number)
elif operator == "/":
    print("Result:", first_number / second_number)
else:
    print("Invaled operator")

Invaled operator


Excellent work again, Abid! You’re clearly understanding the concepts well.

#### Exercise 6 Review (Positive / Negative / Zero)

Your code:

```python
number = int(input("Enter your number: "))

if number > 0:
    print("Positive number")
elif number < 0:
    print("Negative number")
else:
    print("Zero number")
```

✔ Perfect logic
✔ Correct use of `if / elif / else`
✔ Correct conditions

This is exactly how a programmer would write it. Great job 👌

#### Mini Calculator Review

Your code:

```python
first_number = int(input("Enter first number: "))
operator = input("Enter single operator: ") # Operator (+ , - , * , /)
second_number = int(input("Enter second number: "))

if operator == "+":
    print("Result:", first_number + second_number)
elif operator == "-":
    print("Result:", first_number - second_number)
elif operator == "*":
    print("Result:", first_number * second_number)
elif operator == "/":
    print("Result:", first_number / second_number)
else:
    print("Invalid operator")
```

✔ Correct use of `if / elif / else`
✔ Correct comparisons
✔ Correct math operations

Very good logic 👏

### 🚀 Step 8: Loops (Very Powerful)

Loops let a program **repeat actions automatically**.

#### Exercise 7

Write a program that prints numbers **1 to 10** using a loop.

Expected output:

```
1
2
3
4
5
6
7
8
9
10
```

#### 🎯 Bonus Challenge

Print the **multiplication table of 5**.

Example:

```
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
...
5 x 10 = 50
```

Hint:

```
for i in range(1, 11):
```


In [27]:
# Exercise 7
for i in range(1,11):
    print(i)

1
2
3
4
5
6
7
8
9
10


In [33]:
# Bouns Challenge
for i in range(1, 11):
    print(5, "x", i, "=", i * 5)

5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50


Great work again, Abid! 👏🔥 You’re doing **exactly what a good beginner should do: write code and practice**.

#### Exercise 7 Review

Your code:

```python
for i in range(1,11):
    print(i)
```

✔ Correct loop
✔ Correct range
✔ Correct output

This will print numbers **1 → 10**, exactly as required. Perfect ✅

#### Bonus Challenge Review (Multiplication Table)

Your code:

```python
for i in range(1, 11):
    print(5, "x", i, "=", i * 5)
```

✔ Correct loop
✔ Correct multiplication logic
✔ Correct formatting

Output will look like:

```
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
...
5 x 10 = 50
```

Excellent 👍


### 🚀 Step 9: `while` Loops

A **while loop runs as long as a condition is true**.

### Important part

```python
i += 1
```

This increases the number each loop.
Without it, the loop would run **forever** (infinite loop).

#### Exercise 8

Write a program that prints numbers **1 to 5 using a `while` loop**.

Expected output:

```
1
2
3
4
5
```

#### 🎯 Mini Challenge (Loop + Logic)

Make a program that:

1. Starts with number `10`
2. Prints numbers **10 → 1**
3. Then prints `"Blast off!"`

Expected output:

```
10
9
8
7
6
5
4
3
2
1
Blast off!
```

In [40]:
# Exercise 8
i = 1

while i <= 5:
    print(i)
    i += 1

1
2
3
4
5


In [36]:
# Mini Challenge
for i in range(10, 0, -1):
    print(i)
    
print("Blast off!")

10
9
8
7
6
5
4
3
2
1
Blast off!


Excellent, Abid! 🔥 You’re really getting comfortable with Python now.

#### Exercise 8 Review (`while` loop)

Your code:

```python
i = 1

while i <= 5:
    print(i)
    i += 1
```

✔ Correct `while` condition
✔ Correct increment (`i += 1`)
✔ Correct output 1 → 5

Perfect solution 👌

#### Mini Challenge Review

Your code:

```python
for i in range(10, 0, -1):
    print(i)

print("Blast off!")
```

✔ Correct countdown
✔ Correct use of `range(start, stop, step)`
✔ Correct step `-1`
✔ Correct final message

Output will be:

```
10
9
8
7
6
5
4
3
2
1
Blast off!
```

Very nice 👍

Even though the challenge was about loops generally, using a **`for` loop** here is actually **cleaner than `while`**, so this is good programming practice.

### 🚀 Step 10: Lists (Super Important)

Lists store **multiple values in one variable**.

Example:

```python
numbers = [1, 2, 3, 4, 5]
```

Another example:

```python
fruits = ["apple", "banana", "orange"]
```

You can access items by **index**.

```python
fruits = ["apple", "banana", "orange"]

print(fruits[0])
print(fruits[1])
```

Output:

```
apple
banana
```

Important rule:

```
Index starts at 0
```

| Index | Value  |
| ----- | ------ |
| 0     | apple  |
| 1     | banana |
| 2     | orange |

---

#### Exercise 9

Create a list of **5 favorite foods**.

Print:

1. The whole list
2. The first food
3. The last food

Hint:

```python
foods[-1]
```

gets the **last item**.

---

#### Bonus Challenge

Create a list:

```python
numbers = [10, 20, 30, 40, 50]
```

Write a loop that prints:

```
10
20
30
40
50
```

(Hint: loop through the list)


In [45]:
# Exercise 9
foods = ["Meal of Wheat", "Rice", "Meat of Goat", "Fresh Fish", "Butter with honey"]
print(foods)
print(foods[0])
print("The last food:", foods[-1])

['Meal of Wheat', 'Rice', 'Meat of Goat', 'Fresh Fish', 'Butter with honey']
Meal of Wheat
The last food: Butter with honey


In [46]:
# Bonus Challenge
numbers = [10, 20, 30, 40, 50]

for i in numbers:
    print(i)

10
20
30
40
50


Great job again, Abid! 👏🔥 Your Python skills are clearly improving.

#### Exercise 9 Review

Your code:

```python
foods = ["Meal of Wheat", "Rice", "Meat of Goat", "Fresh Fish", "Butter with honey"]
print(foods)
print(foods[0])
print("The last food:", foods[-1])
```

✔ Correct list creation
✔ Correct printing of the whole list
✔ Correct first element `foods[0]`
✔ Correct last element `foods[-1]`

Perfect solution 👌

Example output:

```
['Meal of Wheat', 'Rice', 'Meat of Goat', 'Fresh Fish', 'Butter with honey']
Meal of Wheat
The last food: Butter with honey
```

#### Bonus Challenge Review

Your code:

```python
numbers = [10, 20, 30, 40, 50]

for i in numbers:
    print(i)
```

✔ Correct loop through list
✔ Correct syntax
✔ Clean and professional solution

Output:

```
10
20
30
40
50
```

This is exactly how Python programmers loop through lists. Nice work 💪